src/sentry/api/endpoints/group_hashes_split.py

```diff
@@ -0,0 +1,334 @@
+
+        If a group is split up using this endpoint, new events that would have
+        been associated with this group will instead create 1..n new, more
+        "specific" groups according to their hierarchical group hashes.
+
+        For example, let's say you have a group containing all events whose
+        crashing frame was `log_error`, i.e. events are only grouped by one
+        frame. This is not a very descriptive frame to group by. If this
+        endpoint is hit, new events that crash in `log_error` will be sorted
+        into groups that hash by `log_error` and the next (calling) frame.
+
+        In the future this endpoint will move existing events into the new,
+        right groups.
+
+        :pparam string issue_id: the ID of the issue to split up.
+        :auth: required
+        """
+
+        if not features.has(
+            "organizations:grouping-tree-ui", group.project.organization, actor=request.user
+        ):
+
+            return self.respond(
+                {"error": "This project does not have the grouping tree feature"},
+                status=404,
+            )
+
+        hashes = request.GET.getlist("id")
+        for hash in hashes:
+            if not isinstance(hash, str) or len(hash) != 32:
+                return self.respond({"error": "hash does not look like a grouphash"}, status=400)
+
+        for hash in hashes:
+            _split_group(group, hash)
+
+        return self.respond(status=200)
+
+    def delete(self, request, group):
+        """
+        Un-split group(s) into their parent group
+        `````````````````````````````````````````
+
+        This basically undoes the split operation one can do with PUT on this
+        endpoint. Note that this API is not very RESTful: The group referenced
+        here is one of the subgroups created rather than the group that was
+        split up.
+
+        When unsplitting, all other child groups are left intact and can be
+        merged into the parent via regular issue merge.
+
+        In the future this endpoint will, much like for PUT, move existing
+        events of the referenced group into the parent group.
+
+        :pparam string issue_id: the ID of the issue to split up.
+        :auth: required
+        """
+
+        if not features.has(
+            "organizations:grouping-tree-ui", group.project.organization, actor=request.user
+        ):
+
+            return self.respond(
+                {"error": "This project does not have the grouping tree feature"},
+                status=404,
+            )
+
+        hashes = request.GET.getlist("id")
+        for hash in hashes:
+            if not isinstance(hash, str) or len(hash) != 32:
+                return self.respond({"error": "hash does not look like a grouphash"}, status=400)
+
+        for hash in hashes:
+            _unsplit_group(group, hash)
+
+        return self.respond(status=200)
+
+
+class NoHierarchicalHash(Exception):
```

Comment:
Is this exception translated into an error response somewhere?

Reply:
no I consider it an internal error if we ever get a request like that. That's semantically not right but it gives me better monitoring